src/OpenColorIO/transforms/FileTransform.cpp

```diff
@@ -985,10 +985,34 @@ OIIO_ADD_TEST(FileTransform, AllFormats)
     OIIO_CHECK_ASSERT(FormatExtensionFoundByName("spi3d", "spi3d"));
     OIIO_CHECK_ASSERT(FormatExtensionFoundByName("spimtx", "spimtx"));
     OIIO_CHECK_ASSERT(FormatExtensionFoundByName("vf", "nukevf"));
+}
+
+OIIO_ADD_TEST(FileTransform, FormatByIndex)
+{
+    OCIO::FormatRegistry & formatRegistry = OCIO::FormatRegistry::GetInstance();
 
+    for (unsigned int i = 0; i < formatRegistry.getNumRawFormats(); ++i)  {
+        if (i < formatRegistry.getNumFormats(OCIO::FORMAT_CAPABILITY_WRITE)) {
+            OIIO_CHECK_ASSERT(formatRegistry.getFormatNameByIndex(OCIO::FORMAT_CAPABILITY_WRITE, i) != "");
+            OIIO_CHECK_ASSERT(formatRegistry.getFormatExtensionByIndex(OCIO::FORMAT_CAPABILITY_WRITE, i) != "");
+        }
+        else {
+            OIIO_CHECK_ASSERT(formatRegistry.getFormatNameByIndex(OCIO::FORMAT_CAPABILITY_WRITE, i) == "");
+            OIIO_CHECK_ASSERT(formatRegistry.getFormatExtensionByIndex(OCIO::FORMAT_CAPABILITY_WRITE, i) == "");
+        }
+
+        if (i < formatRegistry.getNumFormats(OCIO::FORMAT_CAPABILITY_READ)) {
+            OIIO_CHECK_ASSERT(formatRegistry.getFormatNameByIndex(OCIO::FORMAT_CAPABILITY_READ, i) != "");
+            OIIO_CHECK_ASSERT(formatRegistry.getFormatExtensionByIndex(OCIO::FORMAT_CAPABILITY_READ, i) != "");
+        }
+        else {
+            OIIO_CHECK_ASSERT(formatRegistry.getFormatNameByIndex(OCIO::FORMAT_CAPABILITY_READ, i) == "");
+            OIIO_CHECK_ASSERT(formatRegistry.getFormatExtensionByIndex(OCIO::FORMAT_CAPABILITY_READ, i) == "");
+        }
+    }
```

Comment:
In the unit test, the variable `i` represents the maximum number of FileFormats. And the same variable is also used as an index for the 'read' file formats and the 'write' file formats. However, a file format could only offer a 'read' capability so the indexes ('read' vs. 'write' file formats) could be different...

Reply:
I did a copy & paste of the unit test, and it fails on my Windows machine in debug mode.  OIIO_CHECK_ASSERT() are all comparing pointers instead of strings.

Reply:
Agreed that the test was not clear nor necessarily valid, I updated it to check read and write capabilities independently.